### ISSUE TITLE
kuberneteselbsvcname randomness

### ISSUE THREAD

**Issue by sylr** [Azure/aks-engine#1437]
**Describe the bug**

kuberneteselbsvcname  is changed everytime using the generated command.

**Steps To Reproduce**

`aks-engine generate cluster.json`

**Expected behavior**

I don't know what's the purpose of this value but the fact that it can not be fixed troubles me. Often I end up regenerating and reapplying the _output to change things on VMSS or to upgrade masters and I don't know what could be the implication of this value changing during the life of a cluster.

**AKS Engine version**

HEAD

**Kubernetes version**

**Additional context**

**Comment by jackfrancis** [comment]
Yeah, this is sloppy. I've confirmed this does not seem to be doing anything bad in practice. Our Standard LB test runs an upgrade, and it doesn't seem to be changing the value of the app name (which is what this random var is for). See cluster create:

```
NAMESPACE     NAME                           TYPE           CLUSTER-IP     EXTERNAL-IP      PORT(S)                  AGE     SELECTOR
default       service/kubernetes             ClusterIP      10.0.0.1       <none>           443/TCP                  3m47s   <none>
kube-system   service/elb                    LoadBalancer   10.0.59.93     52.224.137.242   8765:31180/TCP           3m36s   app=8674754679580002136
kube-system   service/elb-udp                LoadBalancer   10.0.173.242   52.226.106.177   8765:30180/UDP           3m36s   app=8674754679580002136
kube-system   service/kube-dns               ClusterIP      10.0.0.10      <none>           53/UDP,53/TCP,9153/TCP   3m33s   k8s-app=kube-dns
kube-system   service/kubernetes-dashboard   NodePort       10.0.0.63      <none>           443:32315/TCP            3m32s   k8s-app=kubernetes-dashboard
kube-system   service/metrics-server         ClusterIP      10.0.137.153   <none>           443/TCP                  3m36s   k8s-app=metrics-server
kube-system   service/tiller-deploy          ClusterIP      10.0.152.28    <none>           44134/TCP                3m36s   app=helm,name=tiller
```

and after upgrade:

```
NAMESPACE     NAME                           TYPE           CLUSTER-IP     EXTERNAL-IP      PORT(S)                  AGE   SELECTOR
default       service/kubernetes             ClusterIP      10.0.0.1       <none>           443/TCP                  50m   <none>
kube-system   service/elb                    LoadBalancer   10.0.59.93     52.224.137.242   8765:31180/TCP           50m   app=8674754679580002136
kube-system   service/elb-udp                LoadBalancer   10.0.173.242   52.226.106.177   8765:30180/UDP           50m   app=8674754679580002136
kube-system   service/kube-dns               ClusterIP      10.0.0.10      <none>           53/UDP,53/TCP,9153/TCP   49m   k8s-app=kube-dns
kube-system   service/kubernetes-dashboard   NodePort       10.0.0.63      <none>           443:32315/TCP            49m   k8s-app=kubernetes-dashboard
kube-system   service/metrics-server         ClusterIP      10.0.137.153   <none>           443/TCP                  50m   k8s-app=metrics-server
kube-system   service/tiller-deploy          ClusterIP      10.0.152.28    <none>           44134/TCP                50m   app=helm,name=tiller
```

Note the `app=` values for the elb services are identical before/after.

Looking more into the implementation...

**Comment by jackfrancis** [comment]
O.K., confirmed that each template generation flow (which includes an `aks-engine upgrade`) changes the value of this param, and its effects, as you point out @sylr. The reason this does not have any practical effect in most scenarios is because the addon is configured to `EnsureExists`. See:

https://github.com/Azure/aks-engine/blob/master/parts/k8s/addons/kubernetesmasteraddons-elb-svc.yaml#L7
https://github.com/Azure/aks-engine/blob/master/parts/k8s/addons/kubernetesmasteraddons-elb-svc.yaml#L24

In other words, when we upgrade a cluster, the addon spec changes (verified by looking at `/etc/kubernetes/addons/elb-svc.yaml` on a master node before and after upgrade). I'm unsure exactly what the effects of actually reconciling this change would be (they won't be positive) without more repro, but we should simply fix this surface area so that the app selector value for both of these elb services doesn't change unnecessarily.

**Comment by jackfrancis** [comment]
@CecileRobertMichon @mboersma FYI

**Comment by jackfrancis** [comment]
keep open

**Comment by ritazh** [comment]
I think we can close this issue as the kuberneteselbsvcname svc workaround was removed as part of #1622
